plugins/github/komand_github/actions/search/action.py

```diff
@@ -1,26 +1,47 @@
-import komand
 import requests
-from .schema import SearchInput, SearchOutput
+import insightconnect_plugin_runtime
+import urllib.parse
 
+from insightconnect_plugin_runtime.helper import clean
+from insightconnect_plugin_runtime.exceptions import PluginException
+from komand_github.actions.search.schema import SearchInput, SearchOutput, Input, Output, Component
 
-class Search(komand.Action):
+
+class Search(insightconnect_plugin_runtime.Action):
     def __init__(self):
         super(self.__class__, self).__init__(
             name="search",
-            description="Search GitHub for data",
+            description=Component.DESCRIPTION,
             input=SearchInput(),
             output=SearchOutput(),
         )
 
     def run(self, params={}):
+
         try:
-            search_type = params.get("search_type").lower()
-            search_params = {"q": params.get("query")}
-            results = requests.get(
-                "https://api.github.com/search/" + search_type,
-                params=search_params,
-                auth=self.connection.basic_auth,
-            )
-            return {"results": results.json()}
-        except Exception as e:
-            self.logger.error("Search failed. Error: " + str(e))
+            search_type = urllib.parse.quote(params.get(Input.SEARCH_TYPE).lower())
+            query = params.get(Input.QUERY)
+
+            url = requests.compat.urljoin(self.connection.api_prefix, f"/search/{search_type}")
+            search_params = {"q": query}
+
+            results = requests.get(url=url, params=search_params, headers=self.connection.auth_header, timeout=60)
+
+            if results.status_code == 200:
+                return {Output.RESULTS: clean(results.json())}
+
+            else:
+                raise PluginException(
```

Comment:
Can we also pass back data if it's suitable in the errors here which may help customers to debug their issue?

Reply:
I have added all generic error handling to the util file and passing back data